component/src/test/java/io/siddhi/extension/io/cdc/source/TestCaseOfCDCListeningModeMongo.java

```diff
@@ -0,0 +1,327 @@
+
+        //Assert event arrival.
+        Assert.assertTrue(eventArrived.get());
+
+        cdcAppRuntime.shutdown();
+        mongoAppRuntime.shutdown();
+        siddhiManager.shutdown();
+    }
+
+    /**
+     * Test case to Capture Update operations from a Mongo table.
+     */
+    @Test
+    public void testUpdateCDC() throws InterruptedException {
+        log.info("------------------------------------------------------------------------------------------------");
+        log.info("CDC TestCase: Capturing Update change data from Mongo.");
+        log.info("------------------------------------------------------------------------------------------------");
+
+        SiddhiManager siddhiManager = new SiddhiManager();
+
+        String cdcinStreamDefinition = "@app:name('cdcTesting')" +
+                "@source(type = 'cdc'," +
+                " url = '" + replicaSetUri + "'," +
+                " username = '" + username + "'," +
+                " password = '" + password + "'," +
+                " table.name = '" + collectionName + "', " +
+                " operation = 'update', " +
+                " @map(type='keyvalue'))" +
+                "define stream updatestm (id string, amount double);";
+
+        String mongoStoreDefinition = "define stream UpdateStream(name string, amount double);" +
+                "define stream InsertStream(name string, amount double);" +
+                "@Store(type='mongodb', mongodb.uri='" + databaseUri + "')" +
+                "define table SweetProductionTable (name string, amount double);";
+
+        String insertQuery = "@info(name='query3') " +
+                "from InsertStream " +
+                "insert into SweetProductionTable;";
+
+        String updateQuery = "@info(name='queryUpdate') " +
+                "from UpdateStream " +
+                "select name, amount " +
+                "update SweetProductionTable " +
+                "set SweetProductionTable.amount = amount " +
+                "on SweetProductionTable.name == name;";
+
+        SiddhiAppRuntime cdcAppRuntime = siddhiManager.createSiddhiAppRuntime(cdcinStreamDefinition);
+
+        StreamCallback updatingStreamCallback = new StreamCallback() {
+            @Override
+            public void receive(Event[] events) {
+                for (Event event : events) {
+                    currentEvent = event;
+                    eventCount.getAndIncrement();
+                    log.info(eventCount + ". " + event);
+                    eventArrived.set(true);
+                }
+            }
+        };
+
+        cdcAppRuntime.addCallback("updatestm", updatingStreamCallback);
+        cdcAppRuntime.start();
+
+        QueryCallback queryCallback2 = new QueryCallback() {
+            @Override
+            public void receive(long timestamp, Event[] inEvents, Event[] removeEvents) {
+                for (Event event : inEvents) {
+                    log.info("update done: " + event);
```

Comment:
use assert to count the updates and check

Reply:
Assertion added for the events received at the CDC source.